dotCMS/src/main/java/com/dotmarketing/filters/VanityURLFilter.java

```diff
@@ -21,76 +25,110 @@
-            .getInstance();
-    private CMSUrlUtil urlUtil = CMSUrlUtil.getInstance();
+    private final VanityUrlHandlerResolver vanityUrlHandlerResolver;
+    private final CMSUrlUtil urlUtil;
+    private final HostWebAPI hostWebAPI;
+    private final LanguageWebAPI languageWebAPI;
+    private final UserWebAPI  userWebAPI;
+
+    public VanityURLFilter() {
+
+        this (VanityUrlHandlerResolver.getInstance(), CMSUrlUtil.getInstance(),
+                WebAPILocator.getHostWebAPI(), WebAPILocator.getLanguageWebAPI(),
+                WebAPILocator.getUserWebAPI());
+    }
+
+    @VisibleForTesting
+    protected VanityURLFilter(final VanityUrlHandlerResolver vanityUrlHandlerResolver,
+                           final CMSUrlUtil urlUtil,
+                           final HostWebAPI hostWebAPI,
+                           final LanguageWebAPI languageWebAPI,
+                           final UserWebAPI  userWebAPI) {
+
+        this.vanityUrlHandlerResolver = vanityUrlHandlerResolver;
+        this.urlUtil                  = urlUtil;
+        this.hostWebAPI               = hostWebAPI;
+        this.languageWebAPI           = languageWebAPI;
+        this.userWebAPI               = userWebAPI;
+    }
 
     public void init(FilterConfig filterConfig) throws ServletException {
     }
 
-    public void doFilter(ServletRequest req, ServletResponse res,
-            FilterChain filterChain) throws IOException, ServletException {
+    public void doFilter(final ServletRequest req, final ServletResponse res,
+            final FilterChain filterChain) throws IOException, ServletException {
 
-        HttpServletRequest request = (HttpServletRequest) req;
-        HttpServletResponse response = (HttpServletResponse) res;
+        final HttpServletRequest  request  = (HttpServletRequest)  req;
+        final HttpServletResponse response = (HttpServletResponse) res;
 
         //Get the URI from the request and check for possible XSS hacks
-        final String uri = urlUtil.getURIFromRequest(request);
+        final String uri         = this.urlUtil.getURIFromRequest(request);
+        final boolean isFiltered = this.urlUtil.isVanityUrlFiltered (uri);
 
-        //Getting the site form the request
-        Host site;
-        try {
-            site = WebAPILocator.getHostWebAPI().getCurrentHost(request);
-            request.setAttribute("host", site);
-        } catch (Exception e) {
-            Logger.error(this,
-                    String.format("Unable to retrieve current Site from request for URI [%s]",
-                            uri));
-            throw new ServletException(e.getMessage(), e);
-        }
+        if (!isFiltered) {
 
-        //Get the user language
-        long languageId = WebAPILocator.getLanguageWebAPI().getLanguage(request).getId();
+            //Getting the site form the request
+            final Host site = this.getHost(request, uri);
```

Comment:
The call `this.getHost` needs to be outside the if, other filters are expecting to have that site set in the request